src/pymoca/backends/casadi/api.py

```diff
@@ -300,6 +300,14 @@ def load_model(model_folder: str, model_name: str, compiler_options: Dict[str, s
                             and ca.depends_on(metadata[key][i, j], parameter_vector)):
                             setattr(variable, tmp, metadata[key][i, j])
 
+        # Try to coerce parameters into their Python types
+        for p in model.parameters:
+            for attr in CASADI_ATTRIBUTES:
+                v = getattr(p, attr)
+                v_mx = ca.MX(v)
+                if v_mx.is_constant() and v_mx.is_regular():
+                    setattr(p, attr, p.python_type(v))
```

Comment:
Useful, but does this belong in this PR?

Reply:
I just turned this PR into stuff to get JSM to work. And there we need integer parameters to be integers, so this change is needed.

Reply:
Hmm, but can we validly assume integer parameters to evaluate to integers in the CasADi/Python world?

I have my doubts. Consider of this snippet:

```
parameter Integer n = 2;
parameter Integer m = 2 * n +1;
```

clearly `m` is also an integer, but it will be a casadi expression, thereby evaluating to an (integer valued) float.

Reply:
... so I wonder if we shouldn't be casting to integers in the JSM model code instead.

Reply:
Eh, maybe. What this commit does (like with the variable metadata commit) is to make sure the behavior of the _uncached_ model matches that one of the _cached_ one. That's why it's  called "coerce cached ... into their python types".

I'm not sure what to do about the `m` case. When inlining of parameters in Pymoca it should imo end up as `int`. In the general case where it remains an expression until the very end (i.e in the DAE), I do think we want a `ca.floor` around it to make sure it actually is an integer. That said, I'm not sure what the following does or should do.

```
parameter Integer a = 1;
parameter Integer b = a / 2;
```
Does that make `b` zero, or should it raise an error? Either way, I would say this (and the `m` stuff) is a new issue. Again, this commit just makes sure that the cached version matches the uncached/raw version.